source/write/bulk-write.txt

```diff
@@ -0,0 +1,370 @@
+    :language: cpp
+    :dedent:
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+.. _cpp-bulk-start-operation:
+
+Create a Bulk Write Instance
+----------------------------
+
+Before running a bulk write operation, call the ``create_bulk_write()`` method on
+a collection. This method returns an instance of the ``mongocxx::bulk_write``
+class that you can use to store instructions about which types of bulk writes to
+perform.
+
+The following example calls the ``create_bulk_write()`` method on the ``restaurants``
+collection:
+
+.. literalinclude:: /includes/write/bulk-write.cpp
+    :start-after: start-create-bulk-write
+    :end-before: end-create-bulk-write
+    :language: cpp
+    :dedent:
+
+You can then append write models to your ``mongocxx::bulk_write`` instance to define
+the bulk operation. For more information, see the following :ref:`cpp-bulk-define-write-models`
+section.
+
+.. _cpp-bulk-define-write-models:
+
+Define the Write Operations
+---------------------------
+
+For each write operation you want to perform, create an instance of one of
+the following model classes:
+
+- ``mongocxx::model::insert_one``
+- ``mongocxx::model::update_one``
+- ``mongocxx::model::update_many``
+- ``mongocxx::model::replace_one``
+- ``mongocxx::model::delete_one``
+- ``mongocxx::model::delete_many``
+
+Then, append each write model to the ``mongocxx::bulk_write`` instance returned
+by the ``create_bulk_write()`` method.
+
+The following sections show how to create and use instances of the preceding
+write model classes.
+
+.. _cpp-bulk-insert-model:
+
+Insert Operations
+~~~~~~~~~~~~~~~~~
+
+To perform an insert operation, create an instance of the ``mongocxx::model::insert_one``
+class and specify the document you want to insert. Then, append the model instance
+to an instance of the ``mongocxx::bulk_write`` class.
+
+The following example creates an instance of ``mongocxx::model::insert_one`` and appends
+it to a ``mongocxx::bulk_write`` instance called ``bulk``:
+
+.. literalinclude:: /includes/write/bulk-write.cpp
+   :start-after: start-bulk-insert-one
+   :end-before: end-bulk-insert-one
+   :language: cpp
+   :dedent:
+
+To insert multiple documents, create an instance of ``mongocxx::model::insert_many``
+for each document.
+
+.. _cpp-bulk-update-model:
+
+Update Operations
+~~~~~~~~~~~~~~~~~
+
+To update a document, create an instance of ``mongocxx::model::update_one`` and
+pass in the following arguments:
+
+- **Query filter** document, which specifies the criteria used to match documents
+  in your collection.
+- **Update** document, which specifies the kind of update to perform. For more information
+  about update operations, see the :manual:`Field Update Operators
+  </reference/operator/update-field/>` guide in the {+mdb-server+} manual.
+
+This model instructs the driver to update *the first* document that matches your
```

Comment:
Q: Why is this information in the middle of the steps of which arguments to pass in/how to append the model? 

S: Move this information to the beginning of the section about update_one, such as:

To update a single document, create an instance of ``mongocxx::model::update_one`. This model instructs the driver...